src/use-mutation.ts

```diff
@@ -23,6 +23,26 @@ export interface UseMutationOptions<
    */
   keys?: _MutationKeys<TParams, TResult>
 
+  /**
+   * Hook to execute a callback when the mutation is triggered
+   */
+  onMutate?: (...args: TParams) => void
+
+  /**
+   * Hook to execute a callback in case of error
+   */
+  onError?: (error: any) => void
```

Comment:
This should get access to the arguments and the error. I feel like at this point it's better to just have one object:

```suggestion
  onError?: (context: { error: TError, args: TParams } & TContext) => Promise<unknown> | unknown
```

It does have a risk of collision but it should be fine and easier to use than

```suggestion
  onError?: (error: TError, context: TContext, ...args: TParams) => void
```